ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -411,24 +411,36 @@ private IndexValue findKey(StoreKey key, FileSpan fileSpan, IndexEntryType type)
                 true).descendingMap();
         metrics.segmentSizeForExists.update(segmentsMapToSearch.size());
       }
+      int segmentsSearched = 0;
       for (Map.Entry<Offset, IndexSegment> entry : segmentsMapToSearch.entrySet()) {
+        segmentsSearched++;
         logger.trace("Index : {} searching index with start offset {}", dataDir, entry.getKey());
         IndexValue value = entry.getValue().find(key);
         if (value != null) {
           logger.trace("Index : {} found value offset {} size {} ttl {}", dataDir, value.getOffset(), value.getSize(),
               value.getExpiresAtMs());
           if (type.equals(IndexEntryType.ANY)) {
             retValue = value;
+            if(!entry.getValue().isMapped()){
+              metrics.blobFoundInActiveSegmentCount.inc();
+            }
             break;
           } else if (type.equals(IndexEntryType.DELETE) && value.isFlagSet(IndexValue.Flags.Delete_Index)) {
             retValue = value;
+            if(!entry.getValue().isMapped()){
+              metrics.blobFoundInActiveSegmentCount.inc();
+            }
             break;
           } else if (type.equals(IndexEntryType.PUT) && !value.isFlagSet(IndexValue.Flags.Delete_Index)) {
             retValue = value;
+            if(!entry.getValue().isMapped()){
+              metrics.blobFoundInActiveSegmentCount.inc();
+            }
             break;
           }
         }
       }
+      metrics.segmentsAccessedPerBlobCount.inc(segmentsSearched);
```

Comment:
Roughly (atleast for active blobs) this metric would give us a rough idea about the age of requested blobs. Ideally if our we claim that our traffic pattern is inclined towards GETs for recently uploaded blobs, then avg should definitely be 1 or 2 and not near n/2 or less (where n is the total no of segments)